### PR TITLE
fixed 183, ObserveArray.map() doesn't convert jsx to proxies

### PR DESCRIPTION
**Pull request by mikedane**
for #183

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -258,4 +258,16 @@ QUnit.module('@yelm', () => {
 		assert.equal(testInstance.props.foo, 'New Prop Value');
 		assert.equal(divComponent.innerText, 'test New Prop Value');
 	});
+
+	QUnit.test('ObserveArray.map() shouldn\'t convert jsx to proxies', (assert) => {
```
Comment (christopherjbaker) [inline test/test.js]: It's not map specific, actually. Jsx should never be observed. I think this test is to specific.

Comment (mikedane) [reply test/test.js]: What are some other scenarios where JSX would normally be converted? I can add them to the test.

**Comment by christopherjbaker** [comment]
Jsx as props
 `<Foo title={<div>title</div>} />`

I think we should test the thing we changed, not the side effects of that
change. We didn't change .map, we changed how can-observe handles jsx.

File: index.js [hunk 2]
```diff
@@ -6,9 +6,10 @@ import { connect } from './lib/connected-component';
 import { createComponent } from './lib/observable-component';
 import { Component, observer } from './lib/observer-component';
 import * as propertyDecorators from './property-decorators';
+import makeObserve from 'can-observe/src/-make-observe';
+import React from 'react';
```
Comment (justinbmeyer) [inline index.js]: I don't think we want React as a hard dep right?

File: index.js [hunk 3]
```diff
@@ -32,6 +33,15 @@ Object.assign(ylem, {
 	propertyDecorators,
 });
 
+const observe = makeObserve.observe;
+
+makeObserve.observe = function(input) {
```
Comment (justinbmeyer) [inline index.js]: Personally, I think we should just make it possible to turn off 
function return values from being proxied.  Ylem should just extend `observe.Object` and `observe.Array` in a way that this functionality is turned off and then export these things.

I'm not sure `.map()` results such should __ever__ try to turn its results into observables.  `DefineList` does this, but I'm not convinced it's right.  I'm very ok turning this off (for now optionally, later by default) in `can-observe`.

File: test/test.js (hunk 1 above)
Comment (justinbmeyer) [reply test/test.js]:
```
observe( <div/> )
```

should not proxy.

File: index.js (hunk 2 above)
Comment (BigAB) [reply index.js]: We import it into other files in the library too.

It is a peer dependency though

File: index.js (hunk 3 above)
Comment (mikedane) [reply index.js]: I agree, I like the idea of exporting `ObserveObject` and `ObserveArray` with the functionality turned off. While the fix in this PR works, it seems kinda hacky just being in the `index.js` file.

File: test/test.js [hunk 4]
```diff
@@ -259,15 +259,28 @@ QUnit.module('@yelm', () => {
 		assert.equal(divComponent.innerText, 'test New Prop Value');
 	});
 
-	QUnit.test('ObserveArray.map() shouldn\'t convert jsx to proxies', (assert) => {
-		const myList = new ObserveArray([ 1 ]);
+	QUnit.test('should not make jsx observable', (assert) => {
```
Comment (BigAB) [inline test/test.js]: I find this test title a bit misleading, I think it should be something like:
`should not convert React elements into observable objects and arrays' or something similar.

React elements are the things returned by the function calls  that JSX transpile down to.

**Pull request closed without merging**